docs/src/architecture.md

```diff
@@ -1,30 +1,166 @@
+by using them to host "passive" PostgreSQL replica clusters. This is to be
+intended primarily for DR, read-only operations, or cross-region availability -
+even though failovers and promotions in this case must be done manually.
+
+![Example of a multiple Kubernetes cluster architecture distributed over 3 regions each with 3 independent data centers](./images/k8s-architecture-multi.png)
+
+### Single availability zone Kubernetes clusters
+
+In case your Kubernetes cluster has only one availability zone, CloudNativePG
+still provides you with a lot of features to improve HA and DR results for your
+PostgreSQL databases, by pushing out the single point of failure (SPoF) to the
+level of the zone as much as possible.
+
+This scenario is typical of self-managed on-premise Kubernetes clusters, where
+only one data center is available.
+
+Single availability zone Kubernetes is unfortunately the only viable option
+where just **two (2) data centers** are available within reach of a low latency
+connection (normally in the same metropolitan area): this prevents users from
+creating a multi-availability zone Kubernetes cluster (as the minimum number of
+3 zones is not reached) and forces them to create two different Kubernetes
+clusters in an active/passive scenario, where the second is used primarily for
+Disaster Recovery.
+
+![Example of a Kubernetes architecture with only 2 data centers](./images/k8s-architecture-2-az.png)
+
+!!! Hint
+    If you are at en early stage of your Kubernetes journey, please share this
+    document with your infrastructure team. The two data centers setup might
+    be simply the result of a "lift-and-shift" transition to Kubernetes
+    from a traditional bare-metal or VM based infrastructure, and the benefits
+    that Kubernetes offers in a 3+ zone scenario might not have been known,
+    or addressed at the time the infrastructure architecture was designed.
+    Ultimately, a third physical location connected to the other two might
+    represent a valid option to consider for organization, as it reduces the
+    overall costs of the infrastructure by moving the day-to-day complexity
+    from the application level down to the physical infrastructure level.
+
+Please refer to the ["PostgreSQL architecture"](#postgresql-architecture)
+section below for details on how you can design your PostgreSQL clusters within
+your single availability zone Kubernetes cluster through shared-nothing
+deployments at the storage and worker node levels only. For HA, in such a
+scenario it becomes even more important that the PostgreSQL instances are
+located on different worker nodes and do not share the same storage.
+
+For DR, you can push the SPoF out by taking advantage of additional
+[Kubernetes clusters](#deployments-across-kubernetes-clusters), and use them to
+host "passive" PostgreSQL replica clusters. As with other Kubernetes workloads in
+this scenario, promotion of a Kubernetes cluster as primary must be done
```

Comment:
promotion of a CloudNativePG cluster as primary (not Kubernetes cluster) ... right?

Reply:
Nope, this is exactly the use case that I am trying to cover with this page, considering how frequent we are seeing 2 data centre setups. The only possibility to have Postgres running as primary on the other Kubernetes cluster in a 2 data centre setup with two separate K8s clusters, is to promote the passive one as primary and redirect users to that K8s cluster. The aftermath for our operator is that users will be forced to manually promote the designated primary, while local applications will talk to it. This is primarily a DR operation, which means that the other K8s cluster will need heavy manual intervention, including restoring all the replica clusters.

This is very very bad and time consuming on the operational side and we should warn users from Day 0, when the K8s infrastructure is designed. I hope this clarifies.